compass/landice/mesh.py

```diff
@@ -290,18 +288,30 @@ def set_cell_width(self, section_name, thk, bed=None, vx=None, vy=None,
         k = 0.05  # This works well, but could try other values
         spacing_bed = min_spac + (max_spac - min_spac) / (1.0 + np.exp(
             -k * (bed - np.mean([high_bed, low_bed]))))
-        # We only want bed topography to influence spacing within high_dist_bed
-        # from the ice margin. In the region between high_dist_bed and
-        # low_dist_bed, use a linear ramp to damp influence of bed topo.
-        spacing_bed[dist_to_grounding_line >= low_dist_bed] = (
-            (1.0 - (dist_to_grounding_line[
-                dist_to_grounding_line >= low_dist_bed] -
-                low_dist_bed) / (high_dist_bed - low_dist_bed)) *
-            spacing_bed[dist_to_grounding_line >= low_dist_bed] +
-            (dist_to_grounding_line[dist_to_grounding_line >=
-                                    low_dist_bed] - low_dist_bed) /
-            (high_dist_bed - low_dist_bed) * max_spac)
-        spacing_bed[dist_to_grounding_line >= high_dist_bed] = max_spac
+
+        # If max_res_in_ocn is true, use the minimum cell spacing for
+        # all ocean cells. Important for ocean-coupled runs where
+        # resolving fjords and ocean bathymetry is necessary for
+        # thermal forcing parameterizations. Otherwize, telescope out
+        # to coarse resolution with distance from the ice front.
+        if section.get('max_res_in_ocn') == 'True':
+            spacing_bed[np.logical_and(bed < 0, thk == 0)] = min_spac
+            print("Maximizing resolution in ocean. {} ocean cells \
+                  ".format(np.sum(np.logical_and(bed < 0, thk == 0))))
+        else:
```

Comment:
This should not be within an else-block, or we will not be able to set maximum resolution in the ocean _and_ have bed topography influence resolution for the ice sheet.